jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -453,7 +458,8 @@ def process(self, file_list):
 
             # Find the difference between the current readnoise image and the pipeline readnoise reffile, and record image stats.
             # Sometimes, the pipeline readnoise reffile needs to be cutout to match the subarray.
-            pipeline_readnoise = pipeline_readnoise[self.substrt2-1:self.substrt2+self.subsize2-1, self.substrt1-1:self.substrt1+self.subsize1-1]
+            if readnoise.shape != pipeline_readnoise.shape:
```

Comment:
So, one complication, at least for NIRCam, is that the "readnoise" here varies between amplifiers (typically low, ~1 DN, but can be higher), so the readnoise value of a given pixel depends on the amp used in the readout, which can vary depending on if using full frame vs subarray.

Another complication are those subarrays without refpix. In this case, the "readnoise" is much higher than in the full-frame since there is no refpix correction. For these subarrays, Karl and I decided to just use the uncal darks when making the reference files for these subarrays, in the meantime, until we get commissioning data.

Based on the above, I imagine we'll eventually make reference files unique to each subarray once we obtain commissioning subarray darks. This monitor, though, is just setup to compare to whatever reffile is used by the pipeline, so it will compare to cutouts of the full-frame if that's what the pipeline uses.